apps/test/unit/lib/levelbuilder/code-docs-editor/ProgrammingClassEditorTest.jsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import {shallow, mount} from 'enzyme';
+import {expect} from '../../../../util/reconfiguredChai';
+import ProgrammingClassEditor from '@cdo/apps/lib/levelbuilder/code-docs-editor/ProgrammingClassEditor';
+import {getStore} from '@cdo/apps/redux';
+import {Provider} from 'react-redux';
+import sinon from 'sinon';
+
+describe('ProgrammingClassEditor', () => {
+  let defaultProps, initialProgrammingClass, fetchSpy;
+
+  beforeEach(() => {
+    initialProgrammingClass = {
+      id: 1,
+      name: 'Painter',
+      key: 'painter',
+      shortDescription: 'This is a short description.',
+      externalDocumentation: 'developer.mozilla.org',
+      content: 'This is a longer description of the code.',
+      syntax: 'Painter()',
+      returnValue: 'none',
+      tips: 'some tips on how to use this class',
+      examples: [{name: 'example 1'}]
+    };
+    defaultProps = {
+      initialProgrammingClass,
+      environmentCategories: [
+        {key: 'circuit', name: 'Circuit'},
+        {key: 'variables', name: 'Variables'},
+        {key: 'canvas', name: 'Canvas'}
+      ],
+      videoOptions: [
+        {
+          key: 'video1',
+          name: 'Video 1'
+        },
+        {
+          key: 'video2',
+          name: 'Video 2'
+        }
+      ]
+    };
+    fetchSpy = sinon.stub(window, 'fetch');
+  });
+
+  afterEach(() => {
+    fetchSpy.restore();
+  });
+
+  it('displays initial values in input fields', () => {
+    const wrapper = shallow(<ProgrammingClassEditor {...defaultProps} />);
+    expect(wrapper.text().includes('Editing Class')).to.be.true;
+
+    // Display name
+    expect(
+      wrapper
+        .find('input')
+        .at(0)
+        .props().value
+    ).to.equal('Painter');
+
+    // Key
+    expect(
+      wrapper
+        .find('input')
+        .at(1)
+        .props().value
+    ).to.equal('painter');
+    expect(
+      wrapper
+        .find('input')
+        .at(1)
+        .props().readOnly
+    ).to.be.true;
+
+    // Category select
+    const categorySelect = wrapper.find('select').at(0);
+    expect(categorySelect.find('option').length).to.equal(4);
+    expect(
+      categorySelect.find('option').map(option => option.props().value)
+    ).to.eql(['', 'circuit', 'variables', 'canvas']);
+
+    // Documentation section
```

Comment:
This is a bit of a nit, but my preference would be to break this one test up into smaller tests. For example this could be `it('displays initial values for documentation section')`, just so that it's easier to read, and if something fails it's easy to determine what failed